jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushLayersStep.java

```diff
@@ -86,9 +92,15 @@
   private PushBlobStep makePushBlobStep(AsyncStep<? extends CachedLayer> cachedLayerStep)
       throws ExecutionException {
     CachedLayer cachedLayer = NonBlockingSteps.get(cachedLayerStep);
+
+    Allocation progressAllocation =
+        parentProgressAllocation.newChild("push layer " + cachedLayer.getDigest(), 1);
+    buildConfiguration.getEventDispatcher().dispatch(new ProgressEvent(progressAllocation, 0));
```

Comment:
Ah, I thought why it did not have the completion event, but now I think `PushBlobStep` will do it (and we need that kind of premise).

If so, I can see that passing down `parentProgressAllocation` to `PushBlobStep` will do the same if we only think about the entire allocation units, while the difference is that you'll have one more depth in the tree, right?

So, it looks like for this child, the allocation should always be `1`, right?

So these have been the questions I had from the start, like whether to pass down the parent or create a child unit with a single allocation, what if you wanted to pass down a node to multiple sequential calls, what if we also want to do some progress report at this node, etc.

Anyway, we are here at this point, and I guess we'll move forward with this.

Reply:
Yea, basically the constraint is that each step creates its `Allocation` knowing how many children it will have, and passes down the `Allocation` to the children, which each parent does not know how many allocation units each of its children's `Allocation` have.